metadata_test.go

```diff
@@ -0,0 +1,99 @@
+package metadata
+
+import (
+	"testing"
+
+	cidlink "github.com/ipld/go-ipld-prime/linking/cid"
+	"github.com/storacha/go-metadata/internal/testutil"
+	"github.com/stretchr/testify/require"
+)
+
+func TestRoundTripLocationCommitmentMetadata(t *testing.T) {
+	t.Run("all fields", func(t *testing.T) {
+		claim := testutil.RandomCID(t).(cidlink.Link).Cid
+		shard := testutil.RandomCID(t).(cidlink.Link).Cid
+		var length uint64 = 138
```

Comment:
[np] another way of expressing this, that might be more common
```suggestion
		length := uint64(138)
```